src/components/PeoplePage.tsx

```diff
@@ -1,33 +1,173 @@
+            .includes(searchQuery.toLowerCase()) ||
+          person.fatherName?.toLowerCase().includes(searchQuery.toLowerCase()),
+      );
+    }
+
+    if (sortField) {
+      updatedPeople.sort((a, b) => {
+        let comparison = 0;
+
+        if (sortField === 'name') {
+          comparison = a.name.localeCompare(b.name);
+        } else if (sortField === 'sex') {
+          comparison = a.sex.localeCompare(b.sex);
+        } else if (sortField === 'born') {
+          comparison = a.born - b.born;
+        } else if (sortField === 'died') {
+          comparison = a.died - b.died;
+        }
+
+        return sortOrder === 'desc' ? -comparison : comparison;
+      });
+    }
+
+    return updatedPeople;
+  }, [people, filter, centuryQuery, searchQuery, sortField, sortOrder]);
 
-          <div className="column">
-            <div className="box table-container">
-              <Loader />
+  const handleFilterGender = (filterGender: 'all' | 'male' | 'female') => {
+    const newSearchParams = new URLSearchParams(searchParams);
 
-              <p data-cy="peopleLoadingError">Something went wrong</p>
+    if (filterGender === 'all') {
+      newSearchParams.delete('filter');
+    } else {
+      newSearchParams.set('filter', filterGender);
+    }
 
-              <p data-cy="noPeopleMessage">There are no people on the server</p>
+    setSearchParams(newSearchParams);
+  };
 
-              <p>There are no people matching the current search criteria</p>
+  const handleSelectedCentury = (century: string | 'all' | null) => {
+    if (century === 'all' || century === null) {
+      const params = new URLSearchParams(searchParams);
 
-              <PeopleTable />
-            </div>
-          </div>
-        </div>
+      params.delete('century');
+      setSearchParams(params);
+
+      return;
+    }
+
+    const isSelected = centuryQuery.includes(century);
+    const updatedCenturies = isSelected
+      ? centuryQuery.filter(item => item !== century)
+      : [...centuryQuery, century];
+
+    const params = new URLSearchParams(searchParams);
+
+    params.delete('century');
+    updatedCenturies.forEach(cent => params.append('century', cent.toString()));
+
+    setSearchParams(params);
+  };
+
+  const handleSearchQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const query = event.target.value;
+    const newSearchParams = new URLSearchParams(searchParams.toString());
+
+    if (query) {
+      newSearchParams.set('search', query);
+    } else {
+      newSearchParams.delete('search');
+    }
+
+    setSearchParams(newSearchParams);
+  };
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+      <div className="block">
+        {people.length > 0 ? (
+          <PeopleTable
+            persons={filteredAndSortedPeople}
+            sortField={sortField}
+            sortOrder={sortOrder}
+          />
+        ) : (
+          <p>There are no people matching the current search criteria</p>
+        )}
```

Comment:
The condition here checks if there are people to display, but it might be more appropriate to check if 'filteredAndSortedPeople' has any entries instead of 'people'. This ensures that the message about no matching criteria is shown correctly when filters are applied.